iit/model_pairs/base_model_pair.py

```diff
@@ -261,19 +283,35 @@ def train(
             wandb.config.update({"method": self.wandb_method})
             wandb.run.log_code() # type: ignore
 
-        for epoch in tqdm(range(epochs)):
-            train_metrics = self._run_train_epoch(train_loader, loss_fn, optimizer)
-            test_metrics = self._run_eval_epoch(test_loader, loss_fn)
-            if scheduler_cls:
-                self.step_scheduler(lr_scheduler, test_metrics)
-            self.test_metrics = test_metrics
-            self.train_metrics = train_metrics
-            self._print_and_log_metrics(
-                epoch, MetricStoreCollection(train_metrics.metrics + test_metrics.metrics), use_wandb
-            )
+    
+        # Set seed before iterating on loaders for reproduceablility.
+        t.manual_seed(training_args["seed"])
```

Comment:
Is it possible to use a generator for loaders like we do for [numpy](https://github.com/evanhanders/iit/blob/18e467e45ecf28ff3b8b33caf0f5416614d9b614/iit/model_pairs/iit_model_pair.py#L51)? I think I used to set this once globally in the training script before- my bad. :(

Reply:
I'm not totally sure? I got this solution [here](https://discuss.pytorch.org/t/reproducibility-dataloader-shuffle-true-using-seeds/173836). It seems like the random operation is set when the dataloader is turned into an iterable, and someone could use torch functions between initializing and training the model pair, which could hinder reproducibility without putting something here.